### PR TITLE
Fix image upload settings

### PR DESCRIPTION
**Pull request by jofmi**
# Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Description

Fixed WriteJSON payload for updateSettings.

The wrong payload to WriteJSON broke the container without throwing an error - this issue is still not solved as I do not know the cause.

# Checklist:

- [ ] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings, neither in my IDE nor in my browser
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by jofmi** [comment]
Hi @lebe1, both errors do not appear in my local setup.

**Comment by lebe1** [comment]
> Using your frontend branch [fix-settings-update](https://github.com/augustin-wien/augustin-frontend/pull/80/files),I have two issues, I would like you to reproduce.
> 
> 1. Uploading an image with a long weird name like `Screenshot from 2023-12-01 10-29-03.png` leads to the error "GET [http://localhost:3000/[object%20File]](http://localhost:3000/%5Bobject%20File%5D) HTTP/1.1"
> 2. Updating to the main item works fine with newspaper and calendar but item "Digitale Zeitung" leads to the error state that no database values are shown in the backoffice. I think this a frontend error of a hardcoded value, which I thought you solved @nanu-c but maybe not in the settingsupdate frontend page?

After a more extended testing I found more details on these issues.
**1. Error of uploading images**
This error only occurs in my Chrome browser not in my Firefox browser. Therefore, I assume this is a Frontend error. The "GET [http://localhost:3000/[object%20File]](http://localhost:3000/%5Bobject%20File%5D) HTTP/1.1" call  is created after selecting an image to be uploaded. @nanu-c do you have any ideas?

**2. Error of changing main items several items**
This error seems to me now as a backend error. In my Chrome browser as well as in my Firefox browser i can reproduce this error after several updating the main item between newspaper, calendar and digital license. Therefore, something still muss be broken in the backend, when not uploading an image during a settings update. @jofmi can you try reproducing this again?

**Comment by lebe1** [comment]
Hi @jofmi and @nanu-c,
this issue is really giving me headaches. I would like to but really having a hard time simply ignoring this state of the system where the database is not connected to the backend and not throwing any errors.\
In the following two videos I could reproduce both problems on Firefox:

[Screencast_Firefox_DB_not_connected.webm](https://github.com/augustin-wien/augustina-backend/assets/38591956/0c8c8666-e0da-4a74-a6b7-8e54c8192ec3)

[Screencast_Firefox_file_path.webm](https://github.com/augustin-wien/augustina-backend/assets/38591956/0b746dc7-f4a8-457c-be3b-2ca42fe4739f)

Here are both errors shown on Chrome:

[Screencast_Chrome_DB_not_connected.webm](https://github.com/augustin-wien/augustina-backend/assets/38591956/a4b9668f-149c-45eb-9836-ad258ad99ab4)

[Screencast_Chrome_file_path.webm](https://github.com/augustin-wien/augustina-backend/assets/38591956/d5e9e9c1-1706-4643-ab3c-a36ae2bc1cdf)

**Comment by nanu-c** [comment]
> **1. Error of uploading images** This error only occurs in my Chrome browser not in my Firefox browser. Therefore, I assume this is a Frontend error. The "GET [http://localhost:3000/[object%20File]](http://localhost:3000/%5Bobject%20File%5D) HTTP/1.1" call is created after selecting an image to be uploaded. @nanu-c do you have any ideas?

This part is now solved here -> https://github.com/augustin-wien/augustina-frontend/pull/92

**Comment by nanu-c** [comment]
And also i can't reproduce the main item change all breaks error  either here.

File: app/handlers/handlers.go [hunk 1]
```diff
@@ -1374,12 +1374,15 @@ func getSettings(w http.ResponseWriter, r *http.Request) {
...
-	path = ""
 
 	// Get file from image field
 	file, header, err := r.FormFile("Logo")
 	if err != nil {
-		return // No file passed, which is ok
+		log.Info("No file passed or file is invalid", err)
+		// Do not return error, as not passing a file is ok
+		// Could be improved by differentiating between not passed and invalid file
+		err = nil
```
Comment (nanu-c) [inline app/handlers/handlers.go:1384]: maybe close the file, too here.

**Review by nanu-c — APPROVED** [review]

File: app/handlers/handlers.go (hunk 1 above)
Comment (jofmi) [reply app/handlers/handlers.go:1384]: this is not possible if there's an error, the file object cannot be closed since it has not been opened

**Comment by jofmi** [comment]
@lebe1 i have tried reproducing again with your video, with no success. i also tried to create a new item like it was already available in your video, which also created no issues.

**Comment by jofmi** [comment]
Maybe connected: https://github.com/jackc/pgx/issues/635

**Comment by jofmi** [comment]
I have added another improvement that closes all rows after a db query and added additional error logs.

**Comment by jofmi** [comment]
Still no errors in the logs with the latest version of this branch?

**Comment by nanu-c** [comment]
The multipart error should send an error 500 to the client and here https://github.com/augustin-wien/augustina-frontend/pull/94 is a frontend branch that corrects the multipart form post.

**Review by lebe1 — APPROVED** [review]

**Pull request merged**